include/lexy/dsl/delimited.hpp

```diff
@@ -57,21 +57,42 @@ struct _del_chars
         {
             reader.bump();
         }
-        // Otherwise, try to match Unicode characters.
         else if constexpr (!std::is_same_v<decltype(CharClass::char_class_match_cp(char32_t())),
                                            std::false_type>)
         {
-            auto result = lexy::_detail::parse_code_point(reader);
-            if (result.error == lexy::_detail::cp_error::success
-                && CharClass::char_class_match_cp(result.cp))
+            // Try to match any code point in default_encoding or byte_encoding.
+            if constexpr (std::is_same_v<typename Reader::encoding, lexy::default_encoding> ||
+                          std::is_same_v<typename Reader::encoding, lexy::byte_encoding>)
             {
-                reader.set_position(result.end);
+                LEXY_ASSERT(reader.peek() != Reader::encoding::eof(),
+                            "EOF should be checked before calling this");
+
+                auto recover_begin = reader.position();
+                auto cp =  static_cast<char32_t>(reader.peek());
+                reader.bump();
+
+                if (!CharClass::char_class_match_cp(cp))
+                {
+                    finish(context, sink, recover_begin);
+                    _recover(context, recover_begin, reader.position());
+                    reader.set_position(reader.position());
```

Comment:
This is also a no-op.

Reply:
Yeah I don't really seem to understand these `parse_code_point` shenanigans. In the branch I wrote, I seem to need to manually save the begin iterator even though, as far as I can tell, basically the same code runs in the branch below which does not need to do that. In `parse_code_point` the reader position is bumped and so `end` _should_ be equal to `reader.position()`. But it isn't :|

Well there's no need to explain :) Thank for your patience though!